ddtrace/internal/remoteconfig/worker.py

```diff
@@ -31,19 +33,53 @@ def get_poll_interval_seconds():
 
 
 class RemoteConfigWorker(periodic.PeriodicService):
+    """Remote configuration worker.
+
+    This implements a finite-state machine that allows checking the agent for
+    the expected endpoint, which could be enabled after the client is started.
+    """
+
     def __init__(self):
         super(RemoteConfigWorker, self).__init__(interval=get_poll_interval_seconds())
         self._client = RemoteConfigClient()
         log.debug("RemoteConfigWorker created with polling interval %d", get_poll_interval_seconds())
+        self._state = self._agent_check
 
-    def periodic(self):
+    def _agent_check(self):
+        # type: () -> None
+        data = agent._healthcheck()
+        if data:
+            endpoints = data.get("endpoints", [])
+            if endpoints and (
+                REMOTE_CONFIG_AGENT_ENDPOINT in endpoints or ("/" + REMOTE_CONFIG_AGENT_ENDPOINT) in endpoints
+            ):
+                self._state = self._online
+                return
+
+        log.warning(
```

Comment:
as this get logged every 5 sec by default. maybe we should only display longer message on the first check only?